vmdb/app/controllers/ems_infra_controller.rb

```diff
@@ -52,7 +52,18 @@ def scaling
           scale_parameters_formatted << {"name" => p.name, "value" => scale_parameters[p.name]}
         end
       end
-      if scale_parameters_formatted.length > 0
+
+      begin
+        # Check if stack is ready to be updated
+        update_ready = @stack.update_ready?(@stack.raw_status.first)
+      rescue => ex
+        log_and_flash_message(_("Unable to initiate scaling, obtaining of status failed: #{ex}"))
+        return
+      end
+
+      if !update_ready
+        add_flash(_("Provider is not ready to be scaled, another operation is in progress."), :error)
```

Comment:
I'm concerned with how many conditionals are already in this method.  It seems like we have lots of pre-validations that should be methods that encapsulate the message/logging and action and return.  I'm probably simplifying this alot but I dont' really understand all of the details here... 

```ruby
begin
  validate_available_hosts # encapsulates if assigned_hosts > infra.hosts.count
  validate_stack_ready     # @stack.update_ready? + returns or raises the useful message
  validate_scale_parameters # same as above, encapsualtes the scale_parameters_formatted work above and below
...

rescue => err
  # log/alert_message 
end
```